cpp/src/gandiva/to_date_functions_holder.h

```diff
@@ -0,0 +1,253 @@
+                                            ::arrow::TimeUnit::SECOND) {}
+
+  static Status Make(const FunctionNode& node, std::shared_ptr<ToDateHolder>* holder) {
+    const std::string function_name("to_date");
+    return ToDateFunctionsHolder<ToDateHolder>::Make(node, holder, function_name);
+  }
+
+  static Status Make(const std::string& sql_pattern, int32_t suppress_errors,
+                     std::shared_ptr<ToDateHolder>* holder) {
+    return ToDateFunctionsHolder<ToDateHolder>::Make(sql_pattern, suppress_errors,
+                                                     holder);
+  }
+};
+
+/// Function Holder for SQL 'to_time'
+class GANDIVA_EXPORT ToTimeHolder : public ToDateFunctionsHolder<ToTimeHolder> {
+ public:
+  ~ToTimeHolder() override = default;
+
+  ToTimeHolder(const std::string& pattern, int32_t suppress_errors)
+      : ToDateFunctionsHolder<ToTimeHolder>(pattern, suppress_errors, false,
+                                            ::arrow::TimeUnit::MILLI) {}
+
+  static Status Make(const FunctionNode& node, std::shared_ptr<ToTimeHolder>* holder) {
+    const std::string function_name("to_time");
+    return ToDateFunctionsHolder<ToTimeHolder>::Make(node, holder, function_name);
+  }
+
+  static Status Make(const std::string& sql_pattern, int32_t suppress_errors,
+                     std::shared_ptr<ToTimeHolder>* holder) {
+    return ToDateFunctionsHolder<ToTimeHolder>::Make(sql_pattern, suppress_errors,
+                                                     holder);
+  }
+};
+
+/// Function Holder for SQL 'to_timestamp'
+class GANDIVA_EXPORT ToTimestampHolder : public ToDateFunctionsHolder<ToTimestampHolder> {
+ public:
+  ~ToTimestampHolder() override = default;
+
+  ToTimestampHolder(const std::string& pattern, int32_t suppress_errors)
+      : ToDateFunctionsHolder<ToTimestampHolder>(pattern, suppress_errors, false,
+                                                 ::arrow::TimeUnit::MILLI) {}
+
+  static Status Make(const FunctionNode& node,
+                     std::shared_ptr<ToTimestampHolder>* holder) {
+    const std::string function_name("to_timestamp");
+    return ToDateFunctionsHolder<ToTimestampHolder>::Make(node, holder, function_name);
+  }
+
+  static Status Make(const std::string& sql_pattern, int32_t suppress_errors,
+                     std::shared_ptr<ToTimestampHolder>* holder) {
+    return ToDateFunctionsHolder<ToTimestampHolder>::Make(sql_pattern, suppress_errors,
+                                                          holder);
+  }
+};
+
+/// Function Holder for SQL 'unix_timestamp'
+class GANDIVA_EXPORT UnixTimestampHolder
+    : public ToDateFunctionsHolder<UnixTimestampHolder> {
+ public:
+  ~UnixTimestampHolder() override = default;
+
+  UnixTimestampHolder(const std::string& pattern, int32_t suppress_errors)
+      : ToDateFunctionsHolder<UnixTimestampHolder>(pattern, suppress_errors, false,
+                                                   ::arrow::TimeUnit::MILLI) {}
```

Comment:
unix_timestamp returns seconds since epoch https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_unix-timestamp

Reply:
Changes applied